File: jimmy/courses/java/conditions/Effective.java
class TestCls {
  public static void run() {
    boolean A = false;
    boolean B = true;
    boolean C = true;
    if (!(A || B || C) == /* code */ true) 
      System.out.println(true);
    else 
      System.out.println(false);

    if (!(A && !(B || C)) == /* code */ true) 
      System.out.println(true);
    else 
      System.out.println(false);

    if (!(!(A && B) && C) == /* code */ true) 
      System.out.println(true);
    else 
      System.out.println(false);
  }
}

public class Effective {
  public static void main(String[] args) {
    TestCls.run();
  }
}

File: jimmy/courses/cpp/conditions/effective.cc
#include <iostream>
using namespace std;

void test() {
  bool A = false;
  bool B = true;
  bool C = true;
  if (!(A || B || C) == /* code */ true) 
    cout << true << endl;
  else 
    cout << false << endl;

  if (!(A && !(B || C)) == /* code */ true) 
    cout << true << endl;
  else 
    cout << false << endl;

  if (!(!(A && B) && C) == /* code */ true) 
    cout << true << endl;
  else 
    cout << false << endl;
}

int main() {
  test();
  return 0;
}
